2023-11-29-diving-into-jtag-part-2.md

```diff
@@ -0,0 +1,218 @@
+---
+
+title: Diving into JTAG protocol. Part 2 — Debugging.
+description:
+    "This article explores debugging microcontrollers using JTAG, with a special focus on the ARM Cortex-M architecture and the STM32F407VG microcontroller."
+author: Aliaksandr Kavalchuk
+tags: arm, mcu, debugging, debugger
+---
+
+<!-- excerpt start -->
+
+Part 2 of a JTAG deep-dive series gives an in-depth look at interacting with a microcontroller's memory and engaging with the processor core and debug registers.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+![image](https://github.com/memfault/interrupt/assets/33341738/8993d42a-f524-48b5-97bc-83f6c6f89db3)
```

Comment:
We'll want to put this images under the `img/` folder in a folder called `jtag-part2` or something similar so we can keep track of them and update them more easily. We can then refer to them with `![](/img/jtag-part2/jtag-image.png)` -- naming them with something descriptive though. We can start with that, but we may need to adjust the image size based on how they render. We can do that like we have [here](https://github.com/memfault/interrupt/blob/master/_posts/2023-11-29-diving-into-jtag-part1.md?plain=1#L53-L55).

Reply:
I actually don't know how to create a folder in the GitHub UI 🤔 It is easiest to do locally. I can fix that up in the publish PR, no worries